flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessage.java

```diff
@@ -276,22 +279,22 @@ protected Object decode(ChannelHandlerContext ctx, ByteBuf in) throws Exception
 
 		final int backlog;
 
-		final boolean isBuffer;
+		final Buffer.DataType bufferDataType;
```

Comment:
we can still retain the `isBuffer` as before, because it is enough now to avoid increasing the header size.